core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/hybrid/HybridRoleManager.java

```diff
@@ -1272,4 +1438,49 @@ public void removeGroupRoleMappingByGroupName(String groupName) throws UserStore
             throw new UserStoreException(errorMessage, e);
         }
     }
+
+    /**
+     * Get Organization Id from Tenant.
+     *
+     * @param tenantId The tenant id.
+     * @return The organization id.
+     * @throws UserStoreException An unexpected exception has occurred.
+     */
+    private String getOrganizationId(int tenantId) throws UserStoreException {
+
+        String organizationId = CarbonConstants.SUPER_TENANT_ORG_ID;
+        RealmService realmService =  UserStoreMgtDSComponent.getRealmService();
+        if (realmService != null) {
+            Tenant tenant;
+            try {
+                tenant = realmService.getTenantManager().getTenant(tenantId);
+                if (tenant != null) {
+                    if (StringUtils.isEmpty(tenant.getAssociatedOrganizationUUID())) {
+                        tenant.setAssociatedOrganizationUUID(UUID.randomUUID().toString());
+                    }
+                    organizationId = tenant.getAssociatedOrganizationUUID();
+                }
+            } catch (org.wso2.carbon.user.api.UserStoreException e) {
+                throw new UserStoreException("Error while loading tenant.", e);
+            }
+        }
+        return organizationId;
+    }
+
+    /**
+     * Check whether the role is using V2 or not.
+     *
+     * @return boolean - true if role is using V2, false otherwise
+     */
+    private boolean isRoleV2Using() {
+        Boolean legacyAuthzRuntime = CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME;
+
+        if (legacyAuthzRuntime != null) {
+            return !legacyAuthzRuntime;
+        }
+
+        String enableLegacyAuthzRuntimeStr = ServerConfiguration.getInstance()
+                .getFirstProperty(ServerConstants.ENABLE_LEGACY_AUTHZ_RUNTIME);
```

Comment:
Is there any difference between `ServerConstants.ENABLE_LEGACY_AUTHZ_RUNTIME` and `CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME` used in line 1476?

Reply:
yes. CarbonConstants one is the actual value and ServerConstants is the config name